### ISSUE TITLE
Illegal instruction with release binaries

### ISSUE THREAD

**Issue by ZedPea** [turtlecoin/turtlecoin#228]
So we're still seeing this on some CPU's on 0.5.0. Good to have somewhere to track it.
Compiling from source always fixes the issue.

From zoidan:

> I get the illegal instruction on 0.5.0 on an iMac and a MacBook Air bother with intel core 2 duos.
> Simplewallet works under 0.5.0, it didn’t under previous releases. TurtleCoind is still broke.

From Rossobimbo:

(Also 0.5.0)
> linux mint 18.3 cinamon
> cinnamon version 3.6.7
> kernal:4.10.0-38 generic
> core 2 duo cpu E6850 3GHz

I'm not sure if 0.4.2 was working for everyone, but Rossobimbo says it was for him. So probably the rocksDB upgrade causing issues again.

**Comment by RocksteadyTC** [comment]
If we're thinking about yet another RocksDB adjustment, should we also think about what our experience might be like on another DB- SQLite, LMDB, Influx, etc.?

**Comment by brandonlehmann** [comment]
It may not be a bad idea.

**Comment by RocksteadyTC** [comment]
Assuming this is the route we could take, changing databases, let's:

1. list all options for DBs, 
2. discard the ones that can't be implemented *easily* and quickly 
3. discard from those the ones who aren't actively maintained or don't have good documentation
4. have a critical discussion about the remaining options. 

I think RocksDB has served us well, however, we may have grown out of its utility for our specific application. Even if not moving to a new DB right away, it would still help to keep our finger on a few DB projects for the sake of familiarity.

**Comment by RocksteadyTC** [comment]
Is this still an issue?

**Comment by RocksteadyTC** [comment]
Please ping me to reopen this if it needs to be reopened.

**Comment by ZedPea** [comment]
@RocksteadyTC Occurring for a user with a Xeon E5-2670 and Ubuntu 16.04.1

**Comment by thinkpol2** [comment]
Can confirm on Centos 7.5.1804 with X5650:

SIGILL: [https://pastebin.mozilla.org/9088547](https://pastebin.mozilla.org/9088547)

```
$ cat /proc/cpuinfo 
processor	: 0
vendor_id	: GenuineIntel
cpu family	: 6
model		: 44
model name	: Intel(R) Xeon(R) CPU           X5650  @ 2.67GHz
stepping	: 2
microcode	: 0x1a
cpu MHz		: 1596.000
cache size	: 12288 KB
physical id	: 1
siblings	: 12
core id		: 0
cpu cores	: 6
apicid		: 32
initial apicid	: 32
fpu		: yes
fpu_exception	: yes
cpuid level	: 11
wp		: yes
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc aperfmperf eagerfpu pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 cx16 xtpr pdcm pcid dca sse4_1 sse4_2 popcnt aes lahf_lm epb tpr_shadow vnmi flexpriority ept vpid dtherm ida arat
bogomips	: 5319.76
clflush size	: 64
cache_alignment	: 64
address sizes	: 40 bits physical, 48 bits virtual
power management:
```

**Comment by thinkpol2** [comment]
Looking deeper this morning. Disassembly around 0x7d8532:

```
  7d8529:	0f 88 6d 01 00 00    	js     7d869c <_ZNSt10_HashtableISsSt4pairIKSsN7rocksdb23OptionsSanityCheckLevelEESaIS4_ENSt8__detail10_Select1stESt8equal_toISsESt4hashISsENS6_18_Mod_range_hashingENS6_20_Default_ranged_hashENS6_20_Prime_rehash_policyENS6_17_Hashtable_traitsILb1ELb0ELb1EEEEC1IPKS4_EET_SL_mRKSB_RKSC_RKSD_RKS9_RKS7_RKS5_+0x1cc>
  7d852f:	dd 5d c8             	fstp   QWORD PTR [rbp-0x38]
  7d8532:	c5 fb 10 45 c8       	vmovsd xmm0,QWORD PTR [rbp-0x38]
  7d8537:	e8 f4 2e d9 ff       	call   56b430 <ceil@plt>
  7d853c:	c5 fb 10 0d 7c 28 26 	vmovsd xmm1,QWORD PTR [rip+0x26287c]        # a3adc0 <_ZTVN5boost16exception_detail10clone_implINS0_19error_info_injectorINS_7bad_getEEEEE+0xc0>
```

VMOVSD is an AVX instruction, so I'd guess that's the problem for me. However, the E5-2670 mentioned earlier does have AVX, so it must be faulting somewhere else.

**Comment by brandonlehmann** [comment]
Interesting... as AVX is disabled by default unless something else picked it up somewhere. To confirm, you don't have that if you build from source following the directions, correct?

**Comment by thinkpol2** [comment]
Correct, no problems when building from 0.6.3 source. I'm building against Boost 1.58.

**Comment by ZedPea** [comment]
Another report with a E3-1230 V2, Ubuntu 16.04

**Comment by ZedPea** [comment]
Got 2 reports with a i3-330M, both on GNU/Linux

**Comment by ftwillms** [comment]
Happening on an E5-2630 v2, Ubuntu 16.04.

**Comment by rwdim** [comment]
Crash happens here too... 

processor       : 15
vendor_id       : AuthenticAMD
cpu family      : 21
model           : 1
model name      : AMD Opteron(tm) Processor 4280
stepping        : 2
microcode       : 0x600063d
cpu MHz         : 3167.806
cache size      : 2048 KB
physical id     : 1
siblings        : 8
core id         : 7
cpu cores       : 4
apicid          : 23
initial apicid  : 23
fpu             : yes
fpu_exception   : yes
cpuid level     : 13
wp              : yes
flags           : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl nonstop_tsc cpuid extd_apicid aperfmperf pni pclmulqdq monitor ssse3 cx16 sse4_1 sse4_2 popcnt aes xsave avx lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs xop skinit wdt lwp fma4 nodeid_msr topoext perfctr_core perfctr_nb cpb hw_pstate retpoline retpoline_amd rsb_ctxsw ssbd amd_ssbd vmmcall arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold
bugs            : fxsave_leak sysret_ss_attrs null_seg spectre_v1 spectre_v2 spec_store_bypass
bogomips        : 5599.77
TLB size        : 1536 4K pages
clflush size    : 64
cache_alignment : 64
address sizes   : 48 bits physical, 48 bits virtual
power management: ts ttp tm 100mhzsteps hwpstate cpb

AVX present.. 
also happens on 6380's

Recompile fixes.

**Comment by jerme404** [comment]
Me too, Ubuntu 16.04, E5-2670 V2.

**Comment by hiddenbits** [comment]
Same problem:

Ver: 0.6.3
S.O: Debian 9.5 and Ubuntu 16.04
Proc: [Intel Xeon X3430](https://ark.intel.com/products/42927/Intel-Xeon-Processor-X3430-8M-Cache-2_40-GHz)

When i execute walletd daemon, just shows illegal instruction message.

I downloaded _tar.gz_ with source code, compile it and used walletd binary for both CLI and GUI versions of the wallet. Now both works fine...
